contracts/test/libs/TxGen.sol

```diff
@@ -335,6 +363,45 @@ library TxGen {
         }
     }
 
+    function aggregatedInstanceGeneration(Transaction memory txn)
+        internal
+        pure
+        returns (Aggregation.Instance memory aggregationInstance)
+    {
+        uint256 n = 0;
+        bytes32[] memory logicRefs = collectLogicRefs(txn.actions);
+        Compliance.Instance[] memory complianceInstances = new Compliance.Instance[](logicRefs.length / 2);
```

Comment:
You assume that `logicRefs.length` is even but you don't check this.
If a transaction with an uneven number of logicRefs ends up in this function, this bug will be hard to find.

Since our testing builds on this, I would like to clean this up